sns_ik_examples/src/ik_tests.cpp

```diff
@@ -225,39 +274,50 @@ void test(ros::NodeHandle& nh, double num_samples_pos, double num_samples_vel,
     fk_solver.JntToCart(JointList[i], end_effector_pose);
     double elapsed = 0;
     result = JointSeed[i];
-    start_time = boost::posix_time::microsec_clock::local_time();
     int cnt = 0;  // keep track of iteration count to enforce max number of iterations
+    start_time = boost::posix_time::microsec_clock::local_time();
     do {
       q = result; // when iterating start with last solution
-      if (nullspace_bias_task)
-        rc = kdl_nso_solver.CartToJnt(q, end_effector_pose, result);
-      else
-        rc = kdl_solver.CartToJnt(q, end_effector_pose, result);
-
+      rc = kdl_solver.CartToJnt(q, end_effector_pose, result);
       diff = boost::posix_time::microsec_clock::local_time() - start_time;
       elapsed = diff.total_nanoseconds() / 1e9;
     } while (rc < 0 && elapsed < timeout && cnt++ < 100);
     total_time += elapsed;
     kdlPos_indivTime.push_back(elapsed);
-    if (rc>=0)
+    fk_solver.JntToCart(result, end_effector_pose_check);
+    bool inPosBounds = in_pos_bounds(result, ll, ul);
+    if (rc>=0 && inPosBounds && Equal(end_effector_pose, end_effector_pose_check, 1e-3))
       success++;
+    if(use_nullspace_bias_task) {
```

Comment:
Please add a blank line and comment here so that it's easier to read what's going on.